src/util/sawyercoding.c

```diff
@@ -429,16 +431,17 @@ static size_t encode_chunk_repeat(const uint8 *src_buffer, uint8 *dst_buffer, si
 
 	// Iterate through remainder of the source buffer
 	for (i = 1; i < length; ) {
-		searchIndex = max(0, i - 32);
+		searchIndex = (i < 32) ? 0 : (i - 32);
 		searchEnd = i - 1;
 
 		bestRepeatCount = 0;
 		for (repeatIndex = searchIndex; repeatIndex <= searchEnd; repeatIndex++) {
 			repeatCount = 0;
 			maxRepeatCount = min(min(7, searchEnd - repeatIndex), length - i - 1);
+			// maxRepeatCount should not exceed length
+			assert(repeatIndex + maxRepeatCount < length);
+			assert(i + maxRepeatCount < length);
 			for (j = 0; j <= maxRepeatCount; j++) {
-				assert(repeatIndex + j < length);
```

Comment:
I do remember, but at the time I didn't expect #2739 to be merged and now I saw some room for improvement.